runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1062,6 +1062,28 @@ TR_J9ServerVM::sameClassLoaders(TR_OpaqueClassBlock * class1, TR_OpaqueClassBloc
 bool
 TR_J9ServerVM::isUnloadAssumptionRequired(TR_OpaqueClassBlock *clazz, TR_ResolvedMethod *methodBeingCompiled)
    {
+   TR_OpaqueClassBlock * classOfMethod = static_cast<TR_ResolvedJ9JITaaSServerMethod *>(methodBeingCompiled)->classOfMethod();
+      {
+      OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
+      auto it_class = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
+      auto it_classOfMethod = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) classOfMethod);
+      if ((it_class != _compInfoPT->getClientData()->getROMClassMap().end()) &&
+          (it_classOfMethod != _compInfoPT->getClientData()->getROMClassMap().end()))
+         {
+         if (clazz == classOfMethod ||
+            (it_class->second.classLoader == getSystemClassLoader()) ||
+            ((it_class->second.classLoader == it_classOfMethod->second.classLoader) &&
+            !isAnonymousClass(clazz)))
```

Comment:
Is `isAnonymousClass` working correctly? Have you tested this? I see it defined as 
`isAnonymousClass(TR_OpaqueClassBlock *j9clazz) { return (J9_ARE_ALL_BITS_SET(((J9Class*)j9clazz)->romClass->extraModifiers, J9AccClassAnonClass)); }`
but dereferencing the j9clazz at the server should cause problems in my opinion.
I know it's safe to use the other definition:
` isAnonymousClass(J9ROMClass *romClass) { return (J9_ARE_ALL_BITS_SET(romClass->extraModifiers, J9AccClassAnonClass)); }`
where you first obtain the romClass from the cache for clazz